src/hdmf/build/manager.py

```diff
@@ -515,66 +554,65 @@ def __get_cls_dict(self, base, addl_fields, name=None, default_name=None):
-                    docval_arg['default'] = getattr(field_spec, 'default_value', None)
-                docval_args.append(docval_arg)
-
-                # auto-initialize arguments not found in superclass
-                if f not in existing_args:
-                    new_args.append(f)
+            if f == 'help':  # pragma: no cover
+                # (legacy) do not add field named 'help' to any part of class object
+                continue
 
-                # add arguments not found in superclass to fields for getter/setter generation
+            if getattr(field_spec, 'quantity', None) in (ZERO_OR_MANY, ONE_OR_MANY):
+                # if its a MultiContainerInterface, also build clsconf
+                clsconf.append(dict(
+                    attr=f,
+                    type=self.__get_type(field_spec),
+                    add='add_{}'.format(f),
+                    get='get_{}'.format(f),
+                    create='create_{}'.format(f)
+                ))
+            else:
+                # if not, add arguments to fields for getter/setter generation
+                dtype = self.__get_type(field_spec)
                 if self.__ischild(dtype):
                     fields.append({'name': f, 'child': True})
                 else:
                     fields.append(f)
 
-        # if spec provides a fixed name for this type, remove the 'name' arg from docval_args so that values cannot
-        # be passed for a name positional or keyword arg
-        if name is not None:  # fixed name is specified in spec, remove it from docval args
-            docval_args = filter(lambda x: x['name'] != 'name', docval_args)
+            # auto-initialize arguments not found in superclass
+            if f not in existing_args:
+                new_args.append(f)
+
+        classdict = dict()
+
+        if len(fields):
+            classdict.update({base._fieldsname: tuple(fields)})
+        if len(clsconf):
+            classdict.update(__clsconf__=clsconf)
+
+        docval_args = self._build_docval(base, addl_fields, name, default_name)
+
+        if len(fields) or name is not None:
+            @docval(*docval_args)
+            def __init__(self, **kwargs):
+                if name is not None:
+                    kwargs.update(name=name)
+                pargs, pkwargs = fmt_docval_args(base.__init__, kwargs)
+                base.__init__(self, *pargs, **pkwargs)  # special case: need to pass self to __init__
 
-        @docval(*docval_args)
-        def __init__(self, **kwargs):
-            if name is not None:
-                kwargs.update(name=name)
-            pargs, pkwargs = fmt_docval_args(base.__init__, kwargs)
-            base.__init__(self, *pargs, **pkwargs)  # special case: need to pass self to __init__
+                for f in new_args:
+                    arg_val = kwargs.get(f, None)
+                    if arg_val is not None:
+                        setattr(self, f, arg_val)
```

Comment:
```suggestion
                    setattr(self, f, arg_val)
```
I'm pretty sure this branch is not necessary. The setter for `AbstractContainer` already checks for None and will not set it if so